src/scip.rs

```diff
@@ -495,20 +480,16 @@ impl ScipPtr {
     }
 
     /// Create solution
-    pub(crate) fn create_sol(&self) -> Result<Solution, Retcode> {
+    pub(crate) fn create_sol(&self) -> Result<*mut SCIP_SOL, Retcode> {
         let mut sol = MaybeUninit::uninit();
         scip_call! { ffi::SCIPcreateSol(self.raw, sol.as_mut_ptr(), std::ptr::null_mut()) }
```

Comment:
Is this function call infallible?

Reply:
Otherwise I would mark it unsafe and specify when it is safe to call.

Reply:
I'm not sure but can't think of a reason why it wouldn't as long as the scipptr is valid.